does not build on Microsoft Visual Studio 14.0

Fails with:

```
cl /Fodrivers\gl_context\glew.windows.tools.64.obj /c drivers\gl_context\glew.c /nologo /DFREETYPE_ENABLED /Z7 /DDEBUG_ENABLED /DDEBUG_MEMORY_ENABLED /DD3D_DEBUG_INFO /Od /MT /Gd /GR /nologo "/IC:\Program Files (x86)\Windows Kits\10\/Include" /DWINDOWS_ENABLED /DRTAUDIO_ENABLED /DWIN32 /DTYPED_METHOD_BIND /DGLES2_ENABLED /DGLEW_ENABLED "/IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "/IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "/IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt" "/IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "/IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\shared" "/IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\um" "/IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\winrt" /I "/IC:\Program Files (x86)\Microsoft DirectX SDK (June 2010)\/Include" -DOC_X86_ASM -DDEBUG_MEMORY_ALLOC -DSCI_NAMESPACE /DMSVC /GR -DMUSEPACK_ENABLED -DOPENSSL_ENABLED -DSQUISH_ENABLED -DVORBIS_ENABLED -DOPUS_ENABLED -DTHEORA_ENABLED -DTHEORALIB_ENABLED -DPNG_ENABLED -DDDS_ENABLED -DPVR_ENABLED -DJPG_ENABLED -DWEBP_ENABLED -DSPEEX_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DETC1_ENABLED -DPNG_ARM_NEON_OPT=0 -DOPENSSL_NO_ASM -DOPENSSL_THREADS -DL_ENDIAN /Icore /Icore\math /Itools /Idrivers /I. /Iplatform\windows /Iplatform\windows\include /Itools\freetype /Itools\freetype\freetype\include /Idrivers\builtin_openssl2 /Idrivers\builtin_zlib\zlib /Idrivers\gl_context\vorbis /Idrivers\builtin_openssl2\crypto /Idrivers\builtin_openssl2\openssl /Idrivers\builtin_openssl2\crypto\evp /Idrivers\builtin_openssl2\crypto\asn1 /Idrivers\builtin_openssl2\crypto\modes
glew.c
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\vcruntime.h(195): error C2371: 'ptrdiff_t': redefinition; different basic types 
C:\projects\godot\drivers/gl_context/glew.h(157): note: see declaration of 'ptrdiff_t'
scons: *** [drivers\gl_context\glew.windows.tools.64.obj] Error 2
scons: building terminated because of errors.
```

full log: https://ci.appveyor.com/project/Marqin/godot/build/job/aoys55g61omd7d5v

This bug is because Godot uses custom `glew.h`. Got patch that fixes this problem but now I have:

```
drivers\opus\celt\bands.c(555): error C2057: expected constant expression 
drivers\opus\celt\bands.c(555): error C2466: cannot allocate an array of constant size 0 
drivers\opus\celt\bands.c(555): error C2133: 'tmp': unknown size 
drivers\opus\celt\bands.c(582): error C2057: expected constant expression 
drivers\opus\celt\bands.c(582): error C2466: cannot allocate an array of constant size 0 
drivers\opus\celt\bands.c(582): error C2133: 'tmp': unknown size 
drivers\opus\celt\bands.c(1385): error C2057: expected constant expression 
drivers\opus\celt\bands.c(1385): error C2466: cannot allocate an array of constant size 0 
drivers\opus\celt\bands.c(1385): error C2133: '_norm': unknown size
```

Godot HEAD build, without any patches:

> C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\vcruntime.h(195): error C2371: 'ptrdiff_t': redefinition; different basic types 
.\drivers/gl_context/glew.h(157): note: see declaration of 'ptrdiff_t'
scons: *** [drivers\gl_context\glew.windows.opt.tools.64.obj] Error 2
scons: building terminated because of errors.
Command exited with code 2

https://ci.appveyor.com/project/Marqin/godot/build/job/33mg1dhbaboybmtf

--- a/drivers/gl_context/glew.h
+++ b/drivers/gl_context/glew.h
@@ -153,8 +153,7 @@ typedef unsigned short wchar_t;
 #endif
 #if !defined(_PTRDIFF_T_DEFINED) && !defined(_PTRDIFF_T_)
 #  ifdef _WIN64
-//typedef __int64 ptrdiff_t;
-typedef long int ptrdiff_t;
+typedef __int64 ptrdiff_t;
 #  else
 typedef _W64 int ptrdiff_t;
 #  endif

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -289,6 +289,8 @@ def configure(env):
                         print "Compiled program architecture will be a 32 bit executable. (forcing bits=32)."
                 else:
                         print "Failed to detect MSVC compiler architecture version... Defaulting to 32bit executable settings (forcing bits=32). Compilation attempt will continue, but SCons can not detect for what architecture this build is compiled for. You should check your settings/compilation setup."                        
+		if env["bits"]=="64":
+			env.Append(CCFLAGS=['/D_WIN64'])
 	else:
 
 		# Workaround for MinGW. See:

--- a/tools/doc/doc_data.cpp
+++ b/tools/doc/doc_data.cpp
@@ -64,7 +64,7 @@ void DocData::merge_from(const DocData& p_data) {
 				// since polymorphic functions are allowed we need to check the type of
 				// the arguments so we make sure they are different.
 				int arg_count = cf.methods[j].arguments.size();
-				bool arg_used[arg_count];
+				bool* arg_used = new bool[arg_count];
 				for (int l = 0; l < arg_count; ++l) arg_used[l] = false;
 				// also there is no guarantee that argument ordering will match, so we
 				// have to check one by one so we make sure we have an exact match

